django/chat/responses.py

```diff
@@ -141,54 +134,76 @@ def summarize_response(chat, response_message):
     target_language = chat.options.summarize_language
     model = chat.options.summarize_model
 
+    llm = OttoLLM(model)
+
+    # TODO: Extracting text from file may incur Azure Document AI costs.
+    # Need to refactor extract_text to create Cost object with correct user and mode.
```

Comment:
It is annoying to do all the refactoring to pass around user and feature into any function that incurs costs. Often the function in which you could actually calculate costs is several layers deep.

I would like to refactor this so that there is some middleware etc. that keeps track of the request.user and feature for each request in a common cache that any function can access. Because Django processes each request in its own thread, this is possible. It is possible to mess up with async functions, and probably Celery also complicates matters.